File: content/search/source/includes/view/code-snippets/java/FacetFieldsQuery.java
import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoCollection;
import com.mongodb.client.MongoDatabase;
import com.mongodb.client.AggregateIterable;
import org.bson.Document;

import java.util.Arrays;
import java.util.List;

public class FacetFieldsQuery {
    public static void main(String[] args) {
        // Connect to your Atlas deployment
        String uri = "<connectionString>";
        try (MongoClient mongoClient = MongoClients.create(uri)) {
            MongoDatabase database =
                mongoClient.getDatabase("sample_airbnb");
            MongoCollection<Document> collection =
                database.getCollection("listings_SearchableTypes");

            // Run the aggregation query on the View
            List<Document> pipeline = Arrays.asList(
                new Document("$searchMeta",
                    new Document("index", "listingsSearchableTypes")
                        .append("facet",
                            new Document("operator",
                                new Document("text",
                                    new Document("path", "summary")
                                        .append("query",
                                            "ocean view")))
                                .append("facets",
                                    new Document("idFacet",
                                        new Document("type", "string")
                                            .append("path", "idString")
                                            .append("numBuckets", 10))
                                        .append("hostFacet",
                                            new Document("type",
                                                "string")
                                                .append("path",
                                                    "superHostString")))))
            );

            AggregateIterable<Document> results =
                collection.aggregate(pipeline);
            for (Document doc : results) {
                System.out.println(doc.toJson());
            }
        }
    }
}

File: content/search/source/includes/view/code-snippets/python/facet-fields-query.py
from pymongo import MongoClient


def query_view():
    # Connect to your Atlas deployment
    uri = "<connectionString>"
    client = MongoClient(uri)

    # Access your database and collection (View)
    database = client["sample_airbnb"]
    collection = database["listings_SearchableTypes"]

    # Run the aggregation query on the View
    pipeline = [
        {
            "$searchMeta": {
                "index": "listingsSearchableTypes",
                "facet": {
                    "operator": {
                        "text": {
                            "path": "summary",
                            "query": "ocean view"
                        }
                    },
                    "facets": {
                        "idFacet": {
                            "type": "string",
                            "path": "idString",
                            "numBuckets": 10
                        },
                        "hostFacet": {
                            "type": "string",
                            "path": "superHostString"
                        }
                    }
                }
            }
        }
    ]

    results = collection.aggregate(pipeline)
    for doc in results:
        print(doc)


query_view()
